setup.sh

```diff
@@ -0,0 +1,395 @@
+#!/bin/bash
+
+# stops the execution if a command or pipeline has an error
+set -e
+
+# Tinkerbell stack Linux setup script
+#
+# See https://tinkerbell.org/setup for the installation steps.
+
+# file to hold all environment variables 
+ENV_FILE=envrc
+
+# color codes
+RED="$(tput setaf 1)"
+GREEN="$(tput setaf 2)"
+YELLOW="$(tput setaf 3)"
+RESET="$(tput sgr0)"
+INFO="${GREEN}INFO:$RESET"
+ERR="${RED}ERROR:$RESET"
+WARN="${YELLOW}WARNING:$RESET"
+
+get_distribution() {
+	lsb_dist=""
+	# Every system that we officially support has /etc/os-release
+	if [ -r /etc/os-release ]; then
+		lsb_dist="$(. /etc/os-release && echo "$ID")"
+	fi
+	# Returning an empty string here should be alright since the
+	# case statements don't act unless you provide an actual value
+	echo "$lsb_dist"
+}
+
+list_network_interfaces() {
+	echo "Following network interfaces found on the system:"
+	ip -o link show | awk -F': ' '{print $2}' | grep '^[e]'
+}
+
+get_tinkerbell_network_interface() {
```

Comment:
It would be ideal if you could first try to read an ENV-var for this value and then only use `read` if nothing was given.

This is so that the script can be run on CI non-interactively.